pkg/cmd/build/build.go

```diff
@@ -63,27 +61,20 @@ func Run(ctx context.Context, namespace, buildKitHost string, isOktetoCluster bo
 	}
 
 	err = solveBuild(ctx, buildkitClient, opt, progress)
+
 	if registry.IsTransientError(err) {
 		log.Yellow("Failed to push '%s' to the registry, retrying ...", tag)
 		success := true
 		err := solveBuild(ctx, buildkitClient, opt, progress)
 		if err != nil {
 			success = false
 		}
+		err = registry.GetErrorMessage(err, tag)
 		analytics.TrackBuildTransientError(buildKitHost, success)
 		return err
 	}
-	if err != nil {
-		imageRegistry, imageTag := registry.GetRegistryAndRepo(tag)
-		if registry.IsLoggedIntoRegistryButDontHavePermissions(err) {
-			err = okErrors.UserError{E: fmt.Errorf("You are not authorized to push image '%s'.", imageTag),
-				Hint: fmt.Sprintf("Please login into the registry '%s' with a user with push permissions to '%s' or use another image.", imageRegistry, imageTag)}
-		}
-		if registry.IsNotLoggedIntoRegistry(err) {
-			err = okErrors.UserError{E: fmt.Errorf("You are not authorized to push image '%s'.", imageTag),
-				Hint: fmt.Sprintf("Login into the registry '%s' and verify that you have permissions to push the image '%s'.", imageRegistry, imageTag)}
-		}
-	}
+
+	err = registry.GetErrorMessage(err, tag)
```

Comment:
are we logging the original error?

Reply:
yes, we log it here: https://github.com/okteto/okteto/pull/1466/files#diff-3d28f1d0455b7f054ebac7695bbbe3d9c758b44b12602f7365d404a86011a49eR73